src/mpcCoreKit.ts

```diff
@@ -476,15 +476,14 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
     if (!this.tKey.metadata.factorPubs) throw new Error("Factor pubs not present");
     const remainingFactors = this.tKey.metadata.factorPubs[this.tKey.tssTag].length || 0;
     if (remainingFactors <= 1) throw new Error("Cannot delete last factor");
-    if (
-      Point.fromTkeyPoint(factorPub).toBufferSEC1(true).toString("hex") ===
-      Point.fromTkeyPoint(getPubKeyPoint(this.state.factorKey)).toBufferSEC1(true).toString("hex")
-    ) {
+    const fpp = Point.fromTkeyPoint(factorPub);
+    const stateFpp = Point.fromTkeyPoint(getPubKeyPoint(this.state.factorKey));
+    if (fpp.equals(stateFpp)) {
       throw new Error("Cannot delete current active factor");
     }
 
     await deleteFactorAndRefresh(this.tKey, factorPub, this.state.factorKey, this.signatures);
-    const factorPubHex = Point.fromTkeyPoint(factorPub).toBufferSEC1(true).toString("hex");
+    const factorPubHex = fpp.toBufferSEC1(true).toString("hex");
```

Comment:
toString("hex", 64)

Reply:
the convert to public key which is 64 bytes for raw , 65 bytes for full, and 33 bytes for compress
converted to hex would be 128, 130, and 66 chars

Reply:
@himanshuchawla009 
`toBufferSEC1(...)` returns a `Buffer`. Its `toString` method does not what you think here. If you call `toString("hex", n)` on a `Buffer`, it will convert the buffer to hex starting from the `n`-th byte.

Moreover, we should avoid using magic numbers like in `toString("hex", 64)`, as pointed out in my comment above.